src/bootman/sysconfig.c

```diff
@@ -237,10 +175,39 @@ SystemConfig *cbm_inspect_root(const char *path, bool image_mode)
         c->prefix = realp;
         c->wanted_boot_mask = 0;
 
-        if (image_mode) {
-                cmb_inspect_root_image(c, realp);
+        if (getenv("CBM_FORCE_LEGACY")) {
+                force_legacy = true;
+        }
+
+        legacy_boot = get_legacy_boot_device(realp);
+        uefi_boot = get_boot_device();
+
+        /* typically /sys, but we forcibly fail this with our tests */
+        fw_path = string_printf("%s/firmware/efi", cbm_system_get_sysfs_path());
+        native_uefi = nc_file_exists(fw_path);
+
+        /*
+         * uefi has precedence over legacy, if we detected both uefi wins
+         * but if force_legacy is set then we honor "users choice"
+         */
+        if (!force_legacy && ((legacy_boot && uefi_boot) || uefi_boot)) {
+                c->wanted_boot_mask = BOOTLOADER_CAP_UEFI;
+        } else if (!native_uefi && (legacy_boot || force_legacy)) {
+                c->wanted_boot_mask = BOOTLOADER_CAP_LEGACY;
```

Comment:
This condition is handled by the final else clause I think and I would prefer less logic when possible.

Reply:
Hrm actually I think we might want to restructure with this consolidation. How about:

```
if (!force_legacy && (uefi_boot || native_uefi)) {
 set uefi
} else if (force_legacy || legacy_boot) {
 set legacy
} else {
 set uefi
}
```

That logic is the simplest to me and says uefi if not force legacy and uefi is detected, otherwise legacy boot if either forced or detected (as uefi wasn't detected) and then uefi by default when none are detected (and force wasn't set).